src/main/kotlin/com/bardsoftware/papeeria/backend/cosmas/CosmasGoogleCloudService.kt

```diff
@@ -29,20 +30,45 @@ import io.grpc.StatusException
  */
 class CosmasGoogleCloudService(private val bucketName: String,
                                private val storage: Storage = StorageOptions.getDefaultInstance().service) : CosmasGrpc.CosmasImplBase() {
+    private val fileBuffer = ConcurrentHashMap<String, ConcurrentHashMap<String, ByteString>>().withDefault { ConcurrentHashMap() }
```

Comment:
Any sense to replace ConcurrentHashMap in generic type with Map ?

Reply:
`fileBuffer = ConcurrentHashMap<String, Map<...>>`? But with usual Map it will die with exception in iteration.
Or it can work, because in default it still will be ConcurrentHashMap? It's strange, because later I can make a mistake and wirte fileBuffer[...] = mutableMapOf()

Reply:
Map is an interface, it is implemented by both HashMap and ConcurrentHashMap. Generic type doesn't care about the concrete implementation.

However, it is a good point about additional protection from error by requiring concurrent data structure. You can replace it with [ConcurrentMap](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentMap.html) interface then

Reply:
ok